src/libs/actions/Report.js

```diff
@@ -663,14 +664,15 @@ function addAction(reportID, text, file) {
  *
  * @param {Number} reportID
  * @param {Number} sequenceNumber
+ * @param {Boolean} ignoreOrder
  */
-function updateLastReadActionID(reportID, sequenceNumber) {
+function updateLastReadActionID(reportID, sequenceNumber, ignoreOrder = false) {
     const currentMaxSequenceNumber = reportMaxSequenceNumbers[reportID];
-    if (sequenceNumber < currentMaxSequenceNumber) {
+    if (!ignoreOrder && sequenceNumber < currentMaxSequenceNumber) {
```

Comment:
Rather than add an `ignoreOrder` flag that basically says "don't run this function" would it make more sense to not call the function in the first place.

Reply:
? It only skips checking the order, though. It runs the rest of the function

Reply:
Hmm maybe we should remove that logic as well.. this logic is confusing and even with the added comment I'm having some trouble understanding in what other contexts this should be used.

Reply:
Sorry, I'm trying to think of a better suggestion for this...

Reply:
Ok so here is my confusion and would like to get @tgolen's thoughts to see if I'm missing something.

I'm not sure I understand why we need to get the last action from the sorted actions here to mark everything as read...

https://github.com/Expensify/Expensify.cash/blob/c5cb5076dc092ee5246e3410e4557fc2d55b4a88/src/pages/home/report/ReportActionsView.js#L266-L278

Here we are looking at the `sequenceNumber` on the last action... but this information is already available to the `actions/Report` and `maxVisibleSequenceNumber` is kind of a misnomer since it has nothing to do with visibility.

We are filtering out the "loading" actions, but that too could be done in the `actions/Report` file I _think_.

If that's true then we wouldn't need an `ignoreOrder` and could make `sequenceNumber` optional and the logic for this method could then end up being

```js
function updateLastReadActionID(reportID, sequenceNumber) {
    const currentMaxSequenceNumber = reportMaxSequenceNumbers[reportID];
    const lastReadSequenceNumber = sequenceNumber 
        ? sequenceNumber 
        : currentMaxSequenceNumber;
    //... set the local last read 
}
```
Not sure if we need to do this change right now, but it seems more intuitive to "set to the most recent `sequenceNumber` or the one specified".

Reply:
I want to dig into this a little so I'm going to pull the branch down and dive into it. I want to make sure I understand everything before offering up further comments on this thread. Thanks for being patient with me!

Reply:
>  it seems more intuitive to "set to the most recent sequenceNumber when none is passed or use the one specified"

I love coming at it from this angle, and I agree it is much more intuitive.

> but that too could be done in the actions/Report file I think.

I briefly looked at this, and I am pretty concerned that changing `reportMaxSequenceNumbers` to not include loading actions has a high risk of breaking things. I would be more comfortable with keeping a separate `reportMaxSequenceNumbersForNonLoadingActions` (lol at that ugly name). 

It makes a lot of sense to me to move this logic from the view into the action file: https://github.com/Expensify/Expensify.cash/blob/c5cb5076dc092ee5246e3410e4557fc2d55b4a88/src/pages/home/report/ReportActionsView.js#L266-L278

Reply:
I'm afraid I'm the one who is lost now.
What do you mean by loading/not loading actions?

Reply:
In the `ReportActionsView.js`, when it calls `updateLastReadActionID`, it is passing the `maxVisibileSequenceNumber`. It gets that number by getting all the report actions and filtering out the ones with `action.loading`. This is what we are referring to.

What I would like to see for a solution is to have `ReportActionsView.js` updated so that all it does is this:
```
    recordMaxAction() {
        updateLastReadActionID(this.props.reportID);
    }
```

and also:
```
onMarkAsUnread={() => updateLastReadActionID(this.props.reportID, item.action.sequenceNumber - 1)}
``` 

All the rest of the logic will then be encapsulated into the `actions/Report.js` file.

Does that help clarify what we are asking for? It doesn't spell out the changes needed in `Report.js`, so let me know if you need help with that as well. I think this will be enough to point you in the right direction though.